prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,432 @@
+            ExtensionJobList jobList = new ExtensionJobList(pageCount);
+            for (int i = offset; i < offset + pageCount; i++) {
+                String jobName = jobNames.get(i);
+                LOG.info("formatting output for job " + jobName);
+                List<Entity> jobEntities = groupedEntities.get(jobName);
+                EntityList entityList = new EntityList(buildEntityElements(fieldSet, jobEntities), jobEntities.size());
+                jobList.addJob(new ExtensionJobList.JobElement(jobName, entityList));
+                LOG.info("added " + jobEntities.size() + " entities: " + entityList.toString());
+            }
+            return jobList;
+        } catch (FalconException | IOException e) {
+            LOG.error("Failed to get extension job list of " + extensionName + ": ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
+        }
+    }
+
+    @GET
+    @Path("instances/{job-name}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public ExtensionInstanceList getInstances(
+            @PathParam("job-name") final String jobName,
+            @QueryParam("start") final String nominalStart,
+            @QueryParam("end") final String nominalEnd,
+            @DefaultValue("") @QueryParam("instanceStatus") String instanceStatus,
+            @DefaultValue("") @QueryParam("fields") String fields,
+            @DefaultValue("") @QueryParam("orderBy") String orderBy,
+            @DefaultValue("") @QueryParam("sortOrder") String sortOrder,
+            @DefaultValue("0") @QueryParam("offset") final Integer offset,
+            @QueryParam("numResults") Integer resultsPerPage,
+            @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+        resultsPerPage = resultsPerPage == null ? getDefaultResultsPerPage() : resultsPerPage;
+        try {
+            List<Entity> entities = getEntityList("", "", "", TAG_PREFIX_EXTENSION_JOB + jobName, "", doAsUser);
+            HashSet<String> fieldSet = new HashSet<>(Arrays.asList(fields.toUpperCase().split(",")));
+            ExtensionInstanceList instances = new ExtensionInstanceList(entities.size());
+            for (Entity entity : entities) {
+                InstancesResult entityInstances = super.getStatus(
+                        entity.getEntityType().name(), entity.getName(), nominalStart, nominalEnd,
+                        null, null, "STATUS:" + instanceStatus, orderBy, sortOrder, offset, resultsPerPage, null);
+                instances.addEntitySummary(new ExtensionInstanceList.EntitySummary(
+                        getEntityElement(entity, fieldSet), entityInstances.getInstances()));
+            }
+            return instances;
+        } catch (FalconException | IOException e) {
+            LOG.error("Error when listing instances of extension job: " + jobName + ": ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
```

Comment:
Shouldn't this be INTERNAL_SERVER_ERROR as error is generated in server code while handling the request?